source/Nevermore/Transient/DbCommandExtensions.cs

```diff
@@ -31,7 +31,7 @@ public static Task<int> ExecuteNonQueryWithRetryAsync(this DbCommand command, Re
         {
             GuardConnectionIsNotNull(command);
             var effectiveCommandRetryPolicy = (commandRetryPolicy ?? RetryPolicy.NoRetry).LoggingRetries(operationName);
-            return effectiveCommandRetryPolicy.ExecuteAction(async () =>
+            return effectiveCommandRetryPolicy.ExecuteActionAsync(async () =>
```

Comment:
Oooh nice catch. Async lambda's are dangerous.

I expected to see the cancellationToken passed through to `ExecuteActionAsync` though. Is that a concern or is it fine because the stuff inside the lambda shares the token anyway?

Reply:
>I expected to see the cancellationToken passed through to `ExecuteActionAsync` though. Is that a concern or is it fine because the stuff inside the lambda shares the token anyway?

Yeah me too. It should be fine because `ExecuteActionAsync` doesn't actually do anything async, no delays or anything, it just immediately retries. I suppose there is potential benefit to taking it and adding in a `ThrowIfCancellationRequested`, I'm still considering whether I want to add that now or leave it for the time being 🤔

Reply:
I decided to make this change in the end, though I've left overloads that don't take a cancellation token around so that this is not a breaking change as this stuff is public.